Js2

LAakerberg opened the pull request:
Last week of working with the JavaScript-2-CA.

Still missing some feature: 
- Search function.
- Update posts function.

UI/UX:
- More user friendly interface as a loading/spinner bar.
- when creating a post you manually have to refresh the page.
- Moving the delete button to profile page instead of detailed page.
- Add document title...

Clean up:
Have to add some more export/import functions were I repeat my code.
Just to make it's clean an easier to read and find etc. problems. 


Added some more comments in my README.md file for this project.
Just as some notes while I´m working. This will later be changed to an more specific detailed information page about the project.

Fermain reviewed (changes requested):
Hi Linus, this is a good submission with some great patterns at work. There are some things that need to be improved before delivery:

1. **JSDocs**, you have annotated your code with JSDocs but they are not quite right. Each doc comment should be tightly coupled to one `function declaration`, however you frequently use them to define `variables` which is not quite right.
2. **Modular JS** (import/export). You mentioned that this is something that you are aware of and this is quite correct. As an exercise, try to make everything in your project exportable **apart from in your main file**. This may seem very difficult at first, but it can be achieved and this will greatly reduce the organisation of your project.
3. UX, when creating a new post for example - nothing happens. You should redirect the user or show them a message confirming success.
4. **Missing functionality**, there are some missing features that are required to pass:
- Delete post
- Edit post
- View individual post by ID

Please let me know when the missing requirements have been addressed as these are most crucial to your success.

LAakerberg commented:
Changes have been added.

- Cleaned up and added JSDocs
- Made bigger changes with Modular JS (Export/Import) Not 100% correct on everyone. But this is something I will need to
training on in the future. (If possible it would be nice to get the repo from live session, just for training material after the JS2 CA)
- Added better UX when changes is made, as login/register/adding post/edit post ex.

- Delete post (OK)
- Edit post (OK)
- View individual post by ID (OK)

LAakerberg commented:
Finally uploaded all changes and improvment!
